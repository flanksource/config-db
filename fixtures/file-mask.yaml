apiVersion: configs.flanksource.com/v1
kind: ScrapeConfig
metadata:
  name: file-mask-scraper
spec:
  file:
    - type: Config
      id: $.id
      name: $.name
      transform:
        mask:
          - selector: config.name == 'Config1'
            jsonpath: $.password
            value: md5sum
          - selector: config.name == 'Config1'
            jsonpath: $.secret
            value: '***'
      paths:
        - fixtures/data/single-config.json

