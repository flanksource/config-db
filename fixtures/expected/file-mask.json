[{
  "last_modified": "0001-01-01T00:00:00Z",
  "config_class": "Config",
  "config_type": "Config",
  "name": "Config1",
  "id": "1",
  "config": {
    "id": 1,
    "name": "Config1",
    "password": "ec6ef230f1828039ee794566b9c58adc",
    "secret": "***"
  }
}]
